plutus-ledger-api/src/PlutusLedgerApi/V3/Contexts.hs

```diff
@@ -398,6 +400,42 @@ instance PlutusTx.Eq ScriptPurpose where
     a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
   _ == _ = Haskell.False
 
+-- | Like `ScriptPurpose` but also contains redeemers, and an optional datum
+-- for spending scripts.
+data ScriptInfo
```

Comment:
Just a suggestion about an alternative name. `Info` isn't very informative IMHO. Feel free to ignore this suggestion if you like the `ScriptInfo` name better.
```suggestion
data ScriptPurposeArgument
```

Reply:
There have been quite a few similar names mentioned :smile: I'll think about it